src/Switch.tsx

```diff
@@ -0,0 +1,358 @@
+              opacity: 0.2;
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.5;
+            }
+          }
+        `
+      } else {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.hoverBg')};
+              border-color: ${get('colors.border.default')};
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.activeBg')};
+            }
+          }
+        `
+      }
+    }
+  }}
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+
+  ${sx}
+  ${sizeVariants}
+`
+
+const hiddenTextStyles: BetterSystemStyleObject = {
+  visibility: 'hidden',
+  height: 0
+}
+
+const Switch: React.FC<SwitchProps> = ({
+  'aria-labelledby': ariaLabelledby,
+  'aria-describedby': ariaDescribedby,
+  defaultOn,
+  disabled,
+  isLoading,
+  on: onProp,
+  onChange,
+  onClick,
+  size,
+  statusLabelPosition,
+  sx: sxProp
+}) => {
+  const isControlled = typeof defaultOn === 'undefined' && typeof onProp !== 'undefined'
+  const {theme, colorScheme} = useTheme()
+  const [onState, setOnState] = useProvidedStateOrCreate<boolean>(onProp, onChange, Boolean(defaultOn))
+  const acceptsInteraction = !disabled && !isLoading
+  const handleToggleClick: MouseEventHandler = useCallback(
+    e => {
+      if (!isControlled) {
+        setOnState(!onState)
+      }
+      onClick && onClick(e)
+    },
+    [onClick, isControlled, onState, setOnState]
+  )
+
+  useEffect(() => {
+    if (onChange && isControlled) {
+      onChange(Boolean(onProp))
+    }
+  }, [onChange, onProp, isControlled])
+
+  return (
+    <Box
+      display="inline-flex"
+      alignItems="center"
+      flexDirection={statusLabelPosition === 'start' ? 'row' : 'row-reverse'}
+      mb={4}
+      sx={sxProp}
+    >
+      {isLoading ? <Spinner size="small" /> : null}
+      <Text
+        color={acceptsInteraction ? 'fg.default' : 'fg.muted'}
+        fontSize={size === 'small' ? 0 : 1}
+        mx={2}
+        aria-hidden="true"
+        sx={{position: 'relative'}}
+      >
+        <Box textAlign="right" sx={onState ? null : hiddenTextStyles}>
+          On
+        </Box>
+        <Box textAlign="right" sx={onState ? hiddenTextStyles : null}>
+          Off
+        </Box>
+      </Text>
+      <SwitchButton
+        onClick={handleToggleClick}
+        aria-labelledby={ariaLabelledby}
+        aria-describedby={ariaDescribedby}
+        aria-checked={onState}
+        aria-disabled={!acceptsInteraction}
+        role="switch"
+        on={onState}
+        size={size}
+        disabled={!acceptsInteraction}
+        isHighContrast={Boolean(colorScheme?.includes('high_contrast'))}
```

Comment:
This feels a little wrong but I'm not I can't think of a better alternative at the moment

Reply:
Yea, this feels hacky, but I couldn't think of anything either. Maybe we could use a shared enum to define theme names, and then I could just check against that?

For example:
```
enum Themes {
  LIGHT = 'light'
  LIGHT_HC = 'light_high_contrast'
  // ...etc
}
```

Reply:
This is no longer an issue now that we're using component primitives.